### PR TITLE
HSEARCH-4133 Capture entity change events to forward them to an SPI

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net/browse/HSEARCH-4133

There's an example implementation of an automatic indexing strategy that forwards indexing events to an external queue: see `org.hibernate.search.util.impl.integrationtest.mapper.orm.automaticindexing.LocalHeapQueueAutomaticIndexingStrategy`. 

@fax4ever this will be useful as a basis for your work in [HSEARCH-4132](https://hibernate.atlassian.net/browse/HSEARCH-4132).

### CODE REVIEW

**Review by fax4ever — APPROVED** [review]
A very nice and extensive job.
I've nothing to say about it.
I'm merging it...

File: mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/bridge/runtime/impl/DocumentRouter.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+	@Override
+	default void close() {
+	}
+
+	DocumentRouteDescriptor currentRoute(Object entityIdentifier, Supplier<? extends E> entitySupplier,
+			DocumentRoutesDescriptor providedRoutes,
+			BridgeSessionContext context);
+
+	DocumentRoutesDescriptor routes(Object entityIdentifier, Supplier<? extends E> entitySupplier,
```
[the remark this answers is not in the document]
Comment (yrodiere) [reply mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/bridge/runtime/impl/DocumentRouter.java:25]: All the routes. `DocumentRoutesDescriptor`, the return type of this method, has both a `currentRoute` and a `previousRoutes` attributes.

**Comment by yrodiere** [comment]
Merged, thanks!

File: mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/bridge/runtime/impl/DocumentRouter.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (fax4ever) [reply mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/bridge/runtime/impl/DocumentRouter.java:25]: Ok thanks for the clarification!

**Pull request merged**